tsdb/querier.go

```diff
@@ -190,7 +190,14 @@ func findSetMatches(pattern string) []string {
 	}
 	escaped := false
 	sets := []*strings.Builder{{}}
-	for i := 4; i < len(pattern)-2; i++ {
+	init := 4
+	end := len(pattern) - 2
+	// If the regex is wrapped in a group we can remove the first and last parentheses
+	if pattern[4] == '(' && pattern[end-1] == ')' {
```

Comment:
Can you use `pattern[init]` rather than `pattern[4]`?

Reply:
good call